flink-state-backends/flink-statebackend-forst/src/main/java/org/apache/flink/state/forst/ForStDBIterRequest.java

```diff
@@ -0,0 +1,263 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.state.forst;
+
+import org.apache.flink.api.common.state.v2.StateIterator;
+import org.apache.flink.core.state.InternalStateFuture;
+import org.apache.flink.runtime.asyncprocessing.StateRequestHandler;
+import org.apache.flink.runtime.asyncprocessing.StateRequestType;
+import org.apache.flink.util.Preconditions;
+
+import org.rocksdb.RocksDB;
+import org.rocksdb.RocksIterator;
+
+import javax.annotation.Nonnull;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * The Iter access request for ForStDB.
+ *
+ * @param <T> The type of value in iterator returned by get request.
+ */
+public class ForStDBIterRequest<T> {
```

Comment:
How about adding type parameters <K,N,UK,UV,T> and making this class abstract? And two abstract method `deserializeElement` and `buildIteratorAndCompleteFuture` are needed. Thus by providing more specific child classes including `ForStDBMapKeyIterator`, `ForStDBMapValueIterator` and `ForStDBMapEntryIterator`, we could avoid too many raw types and unchecked cast.

WDYT?